Adding the global thread pool to the state specific dispatcher examples (#1597)
amethyst/amethyst: book/src/controlling_system_execution/state-specific_dispatcher.md

        dispatcher_builder.add(MovePaddlesSystem, "move_paddles_system", &[]);

        // Build and setup the `Dispatcher`.
        let mut dispatcher = dispatcher_builder
            .with_pool(world.read_resource::<ArcThreadPool>().clone())
            .build();
        dispatcher.setup(&mut world.res);

        self.dispatcher = Some(dispatcher);
    }
}
```

> By default the dispatcher will create its own pool of worker threads to execute systems in but Amethyst's main dispatcher already has a thread pool setup and configured. As reusing it is more efficient, we pull the global pool from the world and attach the dispatcher to it with `.with_pool()`.

Reviewer: ```suggestion
By default, the dispatcher will create its own pool of worker threads to execute systems in, but Amethyst's main dispatcher already has a thread pool setup and configured. As reusing it is more efficient, we pull the global pool from the world and attach the dispatcher to it with `.with_pool()`.
```

> By default the dispatcher will create it's own pool of worker threads to execute states in but Amethyst's main dispatcher already has a thread pool setup and configured. Normally you'll want to reuse that. The global pool is available as a resource so we pull it from the world and attach the dispatcher to it with `.with_pool()`.

Reviewer 2: ```suggestion
By default the dispatcher will create its own pool of worker threads to execute systems in but Amethyst's main dispatcher already has a thread pool setup and configured. As reusing it is more efficient, we pull the global pool from the world and attach the dispatcher to it with `.with_pool()`.
```

Author: Thanks

        dispatcher_builder.add(MovePaddlesSystem, "move_paddles_system", &[]);

        // Build and setup the `Dispatcher`.
        let mut dispatcher = dispatcher_builder

>             .with_pool(world.read_resource::<ArcThreadPool>().clone())

Reviewer 2: Could you add a quick comment explaining what is happening there? Maybe under the code in a quote block, as it is "advanced" stuff.

Author: Sure. I had a little blurb above but it was pretty light. Moved it below and added some more detail. How's that look to you?
